src/lib/media-query/media-observable.ts

```diff
@@ -0,0 +1,95 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+import {Injectable} from '@angular/core';
+import {Observable} from 'rxjs/Observable';
+import {map} from 'rxjs/operators/map';
+import {filter} from 'rxjs/operators/filter';
+
+import {MediaChange} from './media-change';
+import {BreakPointRegistry} from './breakpoints/break-point-registry';
+import {MatchMedia} from './match-media';
+import {BreakPoint} from './breakpoints/break-point';
+import {mergeAlias} from '../utils/add-alias';
+
+@Injectable()
+export class MediaObservable {
```

Comment:
Now the name is probably a misnomer. It's no longer an "Observable". Maybe MediaSource? MediaWatcher? I don't know.

Reply:
In keeping with Material, I think we're going to go with `MediaObserver`